src/Bedrock.Framework/Protocols/ProtocolWriter.cs

```diff
@@ -72,8 +72,10 @@ public async ValueTask WriteManyAsync<TWriteMessage>(IMessageWriter<TWriteMessag
                     return;
                 }
 
-                foreach (var protocolMessage in protocolMessages)
+                var enumerator = protocolMessages.GetEnumerator();
+                foreach(var protocolMessage in protocolMessages)
                 {
+                    cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Undo this as well

Reply:
Is there a particular reason we would not want to cancel at the earliest point we can here? It is possible the list of messages is very long and the user canceled quite some time ago. Seemed like an omission to me.

Reply:
Maybe but it’s unrelated to this PR